brilirs/src/interp.rs

```diff
@@ -83,27 +63,291 @@ where
+      value_store.insert(String::from(dest), Value::Float(args[0] / args[1]));
+    }
+    Feq => {
+      check_asmt_type(&bril_rs::Type::Float, op_type)?;
+      let args = get_args::<f64>(value_store, 2, args)?;
+      value_store.insert(String::from(dest), Value::Bool(args[0] == args[1]));
+    }
+    Flt => {
+      check_asmt_type(&bril_rs::Type::Float, op_type)?;
+      let args = get_args::<f64>(value_store, 2, args)?;
+      value_store.insert(String::from(dest), Value::Bool(args[0] < args[1]));
+    }
+    Fgt => {
+      check_asmt_type(&bril_rs::Type::Float, op_type)?;
+      let args = get_args::<f64>(value_store, 2, args)?;
+      value_store.insert(String::from(dest), Value::Bool(args[0] > args[1]));
+    }
+    Fle => {
+      check_asmt_type(&bril_rs::Type::Float, op_type)?;
+      let args = get_args::<f64>(value_store, 2, args)?;
+      value_store.insert(String::from(dest), Value::Bool(args[0] <= args[1]));
+    }
+    Fge => {
+      check_asmt_type(&bril_rs::Type::Float, op_type)?;
+      let args = get_args::<f64>(value_store, 2, args)?;
+      value_store.insert(String::from(dest), Value::Bool(args[0] >= args[1]));
+    }
+    Call => unreachable!(), // TODO(yati): Why is Call a ValueOp as well?
+    Phi | Alloc | Load | PtrAdd => unimplemented!(),
+  }
+  Ok(())
+}
+
+fn check_num_args(expected: usize, args: &Vec<String>) -> Result<(), InterpError> {
+  if expected != args.len() {
+    Err(InterpError::BadNumArgs(expected, args.len()))
+  } else {
+    Ok(())
+  }
+}
+
+fn check_num_labels(expected: usize, labels: &Vec<String>) -> Result<(), InterpError> {
+  if expected != labels.len() {
+    Err(InterpError::BadNumArgs(expected, labels.len()))
+  } else {
+    Ok(())
+  }
+}
+
+// Returns whether the program should continue running (i.e., if a Return was
+// *not* executed).
+fn execute_effect_op<T: std::io::Write>(
+  op: &bril_rs::EffectOps,
+  args: &Vec<String>,
+  labels: &Vec<String>,
+  curr_block: &BasicBlock,
+  value_store: &HashMap<String, Value>,
+  mut out: T,
+  next_block_idx: &mut Option<usize>,
+) -> Result<bool, InterpError> {
+  use bril_rs::EffectOps::*;
+  match op {
+    Jump => {
+      check_num_args(0, args)?;
+      check_num_labels(1, labels)?;
+      *next_block_idx = Some(curr_block.exit[0]);
+    }
+    Branch => {
+      let bool_args = get_args::<bool>(value_store, 1, args)?;
+      check_num_labels(2, labels)?;
+      let exit_idx = if bool_args[0] { 0 } else { 1 };
+      *next_block_idx = Some(curr_block.exit[exit_idx]);
+    }
+    Return => {
+      out.flush().map_err(|e| InterpError::IoError(Box::new(e)))?;
+      // NOTE: This only works so long as `main` is the only function
+      return Ok(false);
+    }
+    Print => {
+      write!(
+        out,
+        // NOTE: The Bril spec implies print should just output its arguments, with no newline.
+        // However, brili uses console.log, which does add a newline, so we will too
```

Comment:
I know you didn't write this comment, but FWIW, I recently clarified the docs to indicate that it should print a newline.